tuf/api/metadata.py

```diff
@@ -90,12 +91,14 @@ def from_dict(cls, metadata: Mapping[str, Any]) -> 'Metadata':
         else:
             raise ValueError(f'unrecognized metadata type "{_type}"')
 
-        # NOTE: If Signature becomes a class, we should iterate over
-        # metadata['signatures'], call Signature.from_dict for each item, and
-        # pass a list of Signature objects to the Metadata constructor instead.
+        signatures = []
+        for signature in metadata.pop('signatures'):
```

Comment:
Now popping the `signatures` list for consistently destructing the passed dictionary as described in the newly added "Side Effect" docstring section.